generation_scripts/abiotic/climate_data_processing_script.py

```diff
@@ -0,0 +1,227 @@
+dataset["ssrd_Wm-2"] = dataset["ssrd"] / 2592000
+
+
+# In this step, we delete the initial temperature variables (K), precipitation (m), and
+# surface pressure(Pa), surface solar radiation downward (ssrd), 10m u compoenent wind speed (u10) and
+# rename the remaining variables according to the Virtual Ecosystem naming convention.
+dataset_cleaned = dataset.drop_vars(["d2m", "d2m_C", "t2m", "tp", "sp", "ssrd"])
+dataset_renamed = dataset_cleaned.rename(
+    {
+        "sp_kPa": "atmospheric_pressure_ref",
+        "tp_mm": "precipitation",
+        "t2m_C": "air_temperature_ref",
+        "rh2m": "relative_humidity_ref",
+        "u10": "wind_speed_ref",
+        "ssrd_Wm-2": "downward_shortwave_radiation",
+    }
+)
+
+# ---------------------------
+# 4. Add required variables
+# ---------------------------
+# In addition to the variables from the ERA5-Land datasset, a time series of atmospheric
+# CO2 is needed. We add this here as a constant field.
+air_temp_shape = dataset_renamed["air_temperature_ref"].shape
+dataset_renamed["atmospheric_co2_ref"] = xr.DataArray(
+    400 * np.ones(air_temp_shape),
+    dims=dataset_renamed["air_temperature_ref"].dims,
+    coords=dataset_renamed["air_temperature_ref"].coords,
+)
+
+# Mean annual temperature is calculated from the full time series of air temperatures;
+dataset_renamed["mean_annual_temperature"] = dataset_renamed[
+    "air_temperature_ref"
+].mean(dim="time")
+
+# ---------------------------
+# 5. Interpolate to TOML grid (lat/lon from WGS84 grid)
+# ---------------------------
+# Prepare input dims names used in dataset (assumes ERA5 dataset uses 'latitude' and 'longitude')
+# If dataset uses different names, xarray will raise an error.
+# We select method='nearest' to mimic original behaviour; this can be changed later.
+
+data_interp = dataset_renamed.interp(
+    latitude=("y", lat_grid[:, 0]), longitude=("x", lon_grid[0, :]), method="nearest"
+)
+# Note:
+# The Virtual Ecosystem example data is run on a 90 x 90 m grid.
+# This means that some form of spatial downscaling has to be applied to the dataset, for example by spatially
+# interpolating coarser resolution climate data and including the effects of local topography.
+# This is not yet implemented!
+
+# ---------------------------
+# 6. Reformat into VE-style dataset
+# ---------------------------
+# Reformat coords and dims into VE-style dataset:
+#  - dims: y, x, time_index
+# - coords: x (from TOML), y (from TOML), time_index (0..T-1)
+
+ny, nx = lat_grid.shape
+time_len = data_interp.sizes["time"]
+
+# Rename only time -> time_index
+dataset_xyt = (
+    data_interp.rename_dims({"time": "time_index"})
+    .drop_vars({"time"})
+    .assign_coords(
+        {
+            "x": cell_x,  # from TOML
+            "y": cell_y,
+            "time_index": np.arange(0, time_len),
+            "latitude": (("y", "x"), lat_grid),
+            "longitude": (("y", "x"), lon_grid),
+        }
+    )
+)
+
```

Comment:
This might be resolved with the input toml grid definition, but do we still need an offsetting step? Is this captured by providing the centers of the grid cells at the start?